modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/item/selector/ObjectEntryItemSelectorViewDescriptor.java

```diff
@@ -168,8 +169,11 @@ public SearchContainer<ObjectEntry> getSearchContainer()
 			else {
 				Group scopeGroup = _themeDisplay.getScopeGroup();
 
+				DefaultObjectEntryManager defaultObjectEntryManager =
+					(DefaultObjectEntryManager)_objectEntryManager;
```

Comment:
Theoretically, other object definition storage types (proxy) should work in the integration with Info Framework/Page Builder, these other storage types only implement the `ObjectEntryManager` interface. If they reach this code, it could cause problems. 
Can you test this item selector flow with the sample client extension or with Salesforce?
If the flow doesn't go through this code, we could change to something like [this](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/info/item/action/ObjectEntryInfoItemActionExecutor.java#L82-L85), which would be more robust.